client/network/src/service/out_events.rs

```diff
@@ -62,7 +62,7 @@ pub fn channel(name: &'static str, queue_size_warning: i64) -> (Sender, Receiver
 		queue_size: queue_size.clone(),
 		queue_size_warning,
 		warning_fired: false,
-		creation_backtrace: Backtrace::capture(),
+		creation_backtrace: Backtrace::new_unresolved(),
```

Comment:
Can we not use the channel from utils here?

The metrics field is always None or?

Reply:
`Sender` is not accessible on its own, but only via `OutChannels`. `metrics` is inserted when a new `Sender` is added with `OutChannels::push()`:
https://github.com/paritytech/substrate/blob/80082f154829e3f29e1bb5b9d94c1307d7626b8d/client/network/src/service/out_events.rs#L176-L180

Reply:
I've updated the comment to indicate that `metrics` will be initialized.

Reply:
Metrics in `out_events.rs` and `mpsc.rs` seem different, so I don't know how to reuse `utils` version of the channel.

Reply:
Okay ty. At some point we should probably merge both implementations. The metrics are probably not that different.